pkg/server/api/entry/v1/service.go

```diff
@@ -72,7 +73,73 @@ func (s *Service) GetEntry(ctx context.Context, req *entry.GetEntryRequest) (*ty
 }
 
 func (s *Service) BatchCreateEntry(ctx context.Context, req *entry.BatchCreateEntryRequest) (*entry.BatchCreateEntryResponse, error) {
-	return nil, status.Error(codes.Unimplemented, "method BatchCreateEntry not implemented")
+	resp := new(entry.BatchCreateEntryResponse)
+	for _, eachEntry := range req.Entries {
+		e, err := s.createEntry(ctx, eachEntry, req.OutputMask)
+		resp.Results = append(resp.Results, &entry.BatchCreateEntryResponse_Result{
+			Entry:  e,
+			Status: api.StatusFromError(err),
+		})
+	}
+
+	return resp, nil
+}
+
+func (s *Service) createEntry(ctx context.Context, e *types.Entry, outputMask *types.EntryMask) (*types.Entry, error) {
+	log := rpccontext.Logger(ctx)
+
+	entry, err := api.ProtoToRegistrationEntry(e)
+	if err != nil {
+		log.WithError(err).Error("Invalid request: failed to convert entry")
```

Comment:
very small request: can the log messages here include the word createEntry? just because updateEntry and deleteEntry will basically have the same structure and it may be confusing

Reply:
"create entry" will be added as for middleward as a field

Reply:
Right, the middleware will already have added fields for the service and method name to the logger.